git-branchless-lib/src/core/check_out.rs

```diff
@@ -54,6 +55,40 @@ impl Default for CheckOutCommitOptions {
     }
 }
 
+fn maybe_get_branch_name(
+    current_target: Option<String>,
+    oid: Option<NonZeroOid>,
+    repo: &Repo,
+) -> eyre::Result<Option<String>> {
+    let RepoReferencesSnapshot {
+        head_oid,
+        branch_oid_to_names,
+        ..
+    } = repo.get_references_snapshot()?;
+    if (head_oid.is_some() && head_oid == oid) || current_target == head_oid.map(|o| o.to_string())
+    {
+        // Don't try to checkout the branch if we aren't actually checking anything new out.
+        return Ok(current_target);
+    }
+
+    // Determine if the oid corresponds to exactly a single branch. If so,
+    // check that out directly.
+    match oid {
+        Some(oid) => match branch_oid_to_names.get(&oid) {
+            Some(branch_names) => match branch_names.len() {
+                1 => {
+                    // To remove the `refs/heads/` prefix
+                    let name = CategorizedReferenceName::new(branch_names.iter().next().unwrap());
+                    Ok(Some(name.remove_prefix()?))
+                }
+                _ => Ok(current_target),
+            },
+            None => Ok(current_target),
```

Comment:
```suggestion
            Some(branch_names) => match branch_names.iter().exactly_one() {
                Ok(branch_name) => {
                    // To remove the `refs/heads/` prefix
                    let name = CategorizedReferenceName::new(branch_name);
                    Ok(Some(name.remove_prefix()?))
                }
                Err(_) => Ok(current_target),
            },
```

You can use the `itertools` library's `exactly_one` function here to make it a little safer.

Reply:
Good call, done